mountpoint-s3/tests/fuse_tests/prefetch_test.rs

```diff
@@ -0,0 +1,101 @@
+use fuser::BackgroundSession;
+use mountpoint_s3::prefetch::PrefetcherConfig;
+use mountpoint_s3::S3FilesystemConfig;
+use std::fs::OpenOptions;
+use std::io::Read;
+use std::io::Seek;
+use tempfile::TempDir;
+use test_case::test_case;
+
+use crate::fuse_tests::TestClientBox;
+
+// test for checking either prefetching fails or read original object when object is mutated during read.
+fn prefetch_test_etag<F>(creator_fn: F, prefix: &str, request_size: usize, read_size: usize)
+where
+    F: FnOnce(&str, S3FilesystemConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    const OBJECT_SIZE: usize = 1024 * 1024;
+    const SEEK_POS: u64 = 200 * 1024;
+
+    let prefetcher_config = PrefetcherConfig {
+        first_request_size: request_size,
+        ..Default::default()
+    };
+
+    let test_config = S3FilesystemConfig {
+        prefetcher_config,
+        ..Default::default()
+    };
+
+    let (mount_point, _session, mut test_client) = creator_fn(prefix, test_config);
+    let mut read_buf = vec![0u8; OBJECT_SIZE];
+
+    // Make sure there's an existing directory
```

Comment:
probably just delete this comment (we don't care about the directory in this test)